CRM/Gdpr/Form/Forgetme.php

```diff
@@ -23,7 +23,7 @@ public function preProcess() {
 
     // <!-- To DO - check permission -->
 
-    $this->_contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE);
+    $this->_contactID = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE);
```

Comment:
Ah, so this was defining a dynamic property and the `_contactId` was never being used?

Reply:
You know I kinda forget that php allow for dynamically defined class properties, though it is strange why it work only after using the defined propery instead of the danymicly defined one

Reply:
You mean it wasn't working at all before?

Reply:
yes, the content of _contactId  will always be null before my change. when you just mentioned dynmic propery (which I forget that it is a thing php does) made me kinda confused why it wasn't working before but will check it later when I have time and see what's happening here, probably some hidden civicrm magic behind that.